test/test_client.py

```diff
@@ -1277,3 +1382,124 @@ def test_with_session(self):
     session = rq.Session()
     client = TokenClient('url', '123', session=session)
     eq_(session.params['delegation'], '123')
+
+
+class TestSnapshot(_IntegrationTest):
+
+  def test_allow_snapshot(self):
+    self.client._mkdirs('foo')
+    self.client.allow_snapshot('foo')
+
+  def test_allow_snapshot_double(self):
+    self.client._mkdirs('foo')
+    self.client.allow_snapshot('foo')
+
+  def test_disallow_snapshot(self):
+    self.client._mkdirs('foo')
+    self.client.allow_snapshot('foo')
+    self.client.disallow_snapshot('foo')
+
+  def test_disallow_no_allow(self):
+    self.client._mkdirs('foo')
+    self.client.disallow_snapshot('foo')
+
+  @raises(HdfsError)
+  def test_allow_snapshot_not_exists(self):
+    self.client.allow_snapshot('foo')
+
+  @raises(HdfsError)
+  def test_disallow_snapshot_not_exists(self):
+    self.client.disallow_snapshot('foo')
+
+  @raises(HdfsError)
+  def test_allow_snapshot_file(self):
+    self._write('foo', b'hello')
+    self.client.allow_snapshot('foo')
+
+  @raises(HdfsError)
+  def test_disallow_snapshot_file(self):
+    self._write('foo', b'hello')
+    self.client.disallow_snapshot('foo')
+
+  def test_create_delete_snapshot(self):
+    # One cannot test creation and deletion separately, as one cannot
+    # clean HDFS for test isolation if a created snapshot remains
+    # undeleted.
+    self.client._mkdirs('foo')
+    self.client.allow_snapshot('foo')
+    self.client.create_snapshot('foo', 'mysnap')
+    self.client.delete_snapshot('foo', 'mysnap')
+
+  def test_create_snapshot_name(self):
+    self.client._mkdirs('foo')
+    self.client.allow_snapshot('foo')
+    try:
+      snapshot_path = self.client.create_snapshot('foo', 'mysnap')
+      assert snapshot_path.endswith('/foo/.snapshot/mysnap'), repr(snapshot_path)
```

Comment:
Can you switch to a testing utility (e.g. `ok_` or a more string-specific one) instead of `assert`?

Reply:
Done (`assert_regex` seems to be the only option beyond `_ok`).